src/modules/sub-items/sub.items.module.js

```diff
@@ -259,27 +378,62 @@ export default class SubItemsModule extends Component {
      * @memberof SubItemsModule
      */
     afterPriorityUpdated(response) {
-        this.setState(this.updateItemsOrder.bind(this, response.Location));
+        if (this.state.sortClause === 'LocationPriority') {
+            this.discardAllPages();
+            return;
+        }
+
+        this.updateItemLocation(response.Location);
+    }
+
+    updateItemLocation(location) {
+        this.setState((state) => {
+            const { pages } = state;
+            const { itemPage, itemIndex } = this.findItemPositionByLocationId(pages, location.id);
+            const itemPageData = pages[itemPage];
+            const item = itemPageData.items[itemIndex];
+            const updatedItem = deepClone(item);
+
+            updatedItem.location = location;
+
+            const updatedPageItems = [...itemPageData.items];
+
+            updatedPageItems[itemIndex] = updatedItem;
+
+            const updatedPageData = {
+                ...itemPageData,
+                items: updatedPageItems,
+            };
+
+            return {
+                pages: {
+                    ...state.pages,
+                    [itemPage]: updatedPageData,
+                },
+            };
+        });
     }
 
     /**
-     * Updates items order
+     * Finds on what page item is located
+     * and what is the index of the item in page's items list
      *
-     * @method updateItemsOrder
-     * @param {Object} location
-     * @param {Object} state
-     * @returns {Object}
-     * @memberof SubItemsModule
+     * @returns {{itemPage: number, itemIndex: number}}
      */
-    updateItemsOrder(location, state) {
-        const items = state.items;
-        const item = items.find((element) => element.location.id === location.id);
+    findItemPositionByLocationId(pages, locationId) {
+        let itemPage = null;
+        let itemIndex = null;
 
-        item.location = location;
+        for (const [page, pageData] of Object.entries(pages)) {
```

Comment:
Why not `Array.find`?

Reply:
I think I have chosen this because `for of` with `break;` seemed for me more clear than `Object.entries(pages).find(callback)` with callback returning `true` after finding given page.